paasta_tools/tron_tools.py

```diff
@@ -189,7 +196,41 @@ def get_deploy_group(self) -> Optional[str]:
         return self.config_dict.get("deploy_group", None)
 
     def get_cmd(self):
-        return self.config_dict.get("command")
+        command = self.config_dict.get("command")
+        if self.get_executor() == "spark":
+            # Spark expects to be able to write to MESOS_SANDBOX but we don't
+            # use that environment variable so we have to unset it here
+            command = "unset MESOS_DIRECTORY MESOS_SANDBOX; " + command
+        return command
+
+    def get_env(self):
+        env = super().get_env()
+        spark_env = {}
+        if self.get_executor() == "spark":
+            spark_env = get_mesos_spark_env(
+                spark_app_name="tron_spark_{self.get_service()}_{self.get_instance()}",
+                spark_ui_port=pick_random_port(
+                    f"{self.get_service()}{self.get_instance()}".encode()
+                ),
+                mesos_leader=find_mesos_leader(self.get_cluster()),
+                mesos_secret=load_mesos_secret_for_spark(),
+                paasta_cluster=self.get_cluster(),
+                paasta_pool=self.get_pool(),
+                paasta_service=self.get_service(),
+                paasta_instance=self.get_instance(),
+                docker_img=self.get_docker_url(),
+                volumes=format_volumes(
+                    self.get_volumes(load_system_paasta_config().get_volumes())
+                ),
+                user_spark_opts=self.config_dict.get("spark"),
```

Comment:
does this mean that the key in the tronfig for this will just be `spark`?  If so maybe `spark_args` or something would be better?

Reply:
yea that's probably true